examples/speech_recognition/SpeechRecognitionRunner.js

```diff
@@ -0,0 +1,168 @@
+        frameError.sumSquaredError += squared_error;
+        if (error > frameError.maxError) {
+          frameError.maxError = error;
+        }
+        frameError.sumRelError += rel_error;
+        frameError.sumSquaredRelError += squared_rel_error;
+        if (rel_error > frameError.maxRelError) {
+          frameError.maxRelError = rel_error;
+        }
+        if (error > frameError.threshold) {
+          numErrors ++;
+        }
+      }
+    }
+    frameError.rmsError = Math.sqrt(frameError.sumSquaredError / (numRows * numColumns));
+    frameError.sumRmsError += frameError.rmsError;
+    frameError.numErrors = numErrors;
+    return numErrors;
+  }
+
+  _updateScoreError = (frameError, totalError) => {
+    totalError.numErrors += frameError.numErrors;
+    totalError.numScores += frameError.numScores;
+    totalError.sumRmsError += frameError.rmsError;
+    totalError.sumError += frameError.sumError;
+    totalError.sumSquaredError += frameError.sumSquaredError;
+    if (frameError.maxError > totalError.maxError) {
+      totalError.maxError = frameError.maxError;
+    }
+    totalError.sumRelError += frameError.sumRelError;
+    totalError.sumSquaredRelError += frameError.sumSquaredRelError;
+    if (frameError.maxRelError > totalError.maxRelError) {
+      totalError.maxRelError = frameError.maxRelError;
+    }
+  }
+
+  _getReferenceCompareResults = (totalError, framesNum) => {  //framesNum equals to number of frames in one utterance
+    let avgError = totalError.sumError / totalError.numScores;
+    let avgRmsError= totalError.sumRmsError / framesNum;
+    let stdDevError= this._stdDevError(totalError);
+
+    return {
+      maxError: totalError.maxError.toFixed(15),
+      avgError: avgError.toFixed(15),
+      avgRmsError: avgRmsError.toFixed(15),
+      stdDevError: stdDevError.toFixed(15),
+      num: totalError.numErrors
+    }
+  }
+
+  _stdDevError = (totalError) => {
+    let result = Math.sqrt(totalError.sumSquaredError / totalError.numScores
+                - (totalError.sumError / totalError.numScores) * (totalError.sumError / totalError.numScores));
+    return result;
+  }
+
+  run = async (src) => {
+    let status = 'ERROR';
+    let totalTime = 0;
+    let totalError = {};
+    let frameError = {};
+    let referenceTensor = new Float32Array(this._outputTensor[0]);
+
+    let arkInput = await this._getTensorArrayByArk(src);
+    let arkScore = await this._getTensorArrayByArk(this._currentModelInfo.scoreFile)
+    this._initError(totalError);
+
+    for (let i = 0; i < arkInput.rows; i ++) {
+      this._inputTensor[0].set(arkInput.data.subarray(i * arkInput.columns, (i + 1) * arkInput.columns));
+      let start = performance.now();
+      status = await this._model.compute(this._inputTensor, this._outputTensor);
+      let elapsed = performance.now() - start;
+      totalTime += elapsed;
+      referenceTensor.set(arkScore.data.subarray(i * arkScore.columns, (i + 1) * arkScore.columns));
```

Comment:
@miaobin I suggest do this comparing process with computed outputTensor on example side, this runner only does computing with inputTensor, @huningxin @ibelem what's your opinion, thanks.

Reply:
I made changes according to your suggestion and got normal results!
@ibelem The cause of the poor performance has been found and corrected.

Reply:
In fact, this is not a poor performance issue.